helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -433,6 +432,92 @@ public void removeDeploymentGroup(final String name) throws DeploymentGroupDoesN
+      throws DeploymentGroupDoesNotExistException {
+    try {
+      final Node node = client.getNode(Paths.statusDeploymentGroupRollingOps(groupName));
+      return Json.read(node.getBytes(), new TypeReference<List<String>>() {});
+    } catch (JsonMappingException e) {
+      // Can't parse JSON. May as well act as if there's been no rolling operations.
+      return emptyList();
+    } catch (NoNodeException e) {
+      throw new DeploymentGroupDoesNotExistException(groupName, e);
+    } catch (KeeperException | IOException e) {
+      throw new HeliosRuntimeException("getting deployment group rolling-operations failed", e);
+    }
+  }
+
+  private RollingOperation getRollingOperation(final ZooKeeperClient client,
+                                               final String rollingOpId)
+    throws RollingOperationDoesNotExistException {
+    try {
+      final byte[] data = client.getData(Paths.configRollingOp(rollingOpId));
+      return Json.read(data, RollingOperation.class);
+    } catch (NoNodeException e) {
+      throw new RollingOperationDoesNotExistException(rollingOpId);
+    } catch (KeeperException | IOException e) {
+      throw new HeliosRuntimeException("getting rolling-operation " + rollingOpId + " failed", e);
+    }
+  }
+
+  public RollingOperationStatus getRollingOperationStatus(final String rollingOpId)
+      throws RollingOperationDoesNotExistException {
+    final ZooKeeperClient client = provider.get("getRollingOperationStatus");
+    return getRollingOperationStatus(client, rollingOpId);
+  }
+
+  private RollingOperationStatus getRollingOperationStatus(final ZooKeeperClient client,
+                                                           final String rollingOpId)
+      throws RollingOperationDoesNotExistException {
+    try {
+      final byte[] data = client.getData(Paths.statusRollingOp(rollingOpId));
+      return Json.read(data, RollingOperationStatus.class);
+    } catch (NoNodeException e) {
+      throw new RollingOperationDoesNotExistException(rollingOpId);
+    } catch (KeeperException | IOException e) {
+      throw new HeliosRuntimeException("getting rolling-operation " + rollingOpId + " failed", e);
+    }
+  }
+
+  @Override
+  public RollingOperation getLastRollingOperation(final String groupName)
+      throws DeploymentGroupDoesNotExistException {
+    final ZooKeeperClient client = provider.get("getLastRollingOperation");
+    return getLastRollingOperation(client, groupName);
+  }
+
+  private RollingOperation getLastRollingOperation(final ZooKeeperClient client,
+                                                       final String groupName)
+      throws DeploymentGroupDoesNotExistException {
+    // The most recent rolling operations should always be at the head of the list returned by
+    // getRollingOperations().
+    for (final String rollingOpId : getRollingOperationIds(client, groupName)) {
```

Comment:
Is this a style preference, or are there other advantages? I've always felt this way was cleaner. (I think I remember you talking about this with @gimaker recently?)